Symfony/CS/Tests/Fixer/PSR2/UnneededControlParenthesesFixerTest.php

```diff
@@ -0,0 +1,390 @@
+<?php
+
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Tests\Fixer\PSR2;
+
+use Symfony\CS\Fixer\PSR2\UnneededControlParenthesesFixer;
+use Symfony\CS\Tests\Fixer\AbstractFixerTestBase;
+
+/**
+ * @author Sullivan Senechal <soullivaneuh@gmail.com>
+ *
+ * @internal
+ */
+final class UnneededControlParenthesesFixerTest extends AbstractFixerTestBase
+{
+    private static $defaultStatements = null;
+
+    protected function setUp()
+    {
+        // Store default statements to set it with full configuration.
+        if (null === self::$defaultStatements) {
+            $controlStatementsProperty = new \ReflectionProperty('Symfony\CS\Fixer\PSR2\UnneededControlParenthesesFixer', 'controlStatements');
+            $controlStatementsProperty->setAccessible(true);
+            self::$defaultStatements = $controlStatementsProperty->getValue(null);
+        }
+    }
+
+    /**
+     * @dataProvider provideFixCases
+     */
+    public function testFix($expected, $input = null)
+    {
+        // PHP <5.5 BC
+        if (version_compare(PHP_VERSION, '5.5', '<') && false !== strpos($input, 'yield')) {
+            $input = null;
+        }
+
+        // Default config. Fixes all statements.
+        UnneededControlParenthesesFixer::configure(self::$defaultStatements);
+        $this->makeTest($expected, $input);
+
+        // Empty array config. Should not fix anything.
+        UnneededControlParenthesesFixer::configure(array());
+        $this->makeTest($expected, null);
+
+        // Test with only some statements
+        $statements = array('switch_case', 'return');
```

Comment:
Test sample with only one option active (separately for **each** of them): https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/master/Symfony/CS/Tests/Fixer/Contrib/PhpUnitConstructFixerTest.php#L47-L53